src/Npgsql/Replication/NpgsqlReplicationConnection.cs

```diff
@@ -0,0 +1,888 @@
+    /// <see cref="Physical.NpgsqlPhysicalReplicationConnection"/>.
+    /// </summary>
+    public abstract class NpgsqlReplicationConnection : IAsyncDisposable
+    {
+        #region Fields
+
+        static readonly Version FirstVersionWithoutDropSlotDoubleCommandCompleteMessage = new Version(13, 0);
+        static readonly Version FirstVersionWithTemporarySlots = new Version(10, 0);
+        static readonly NpgsqlLogger Log = NpgsqlLogManager.CreateLogger(nameof(NpgsqlReplicationConnection));
+        readonly Stopwatch _cancelTimer = new Stopwatch();
+        readonly NpgsqlConnection _npgsqlConnection;
+        readonly SemaphoreSlim _feedbackSemaphore = new SemaphoreSlim(1, 1);
+        string? _userFacingConnectionString;
+        volatile int _state;
+        TimeSpan? _commandTimeout;
+        TimeSpan _walReceiverTimeout = TimeSpan.FromSeconds(60d);
+        Timer _sendFeedbackTimer;
+        Timer _requestFeedbackTimer;
+        TimeSpan _requestFeedbackInterval;
+
+        // We represent the log sequence numbers as unsigned long
+        // although we have a special struct to represent them and
+        // they are in fact unsigned 64-bit integers, because
+        // we access them via Interlocked to synchronize access
+        // and overcome non-atomic reads/writes on 32-bit platforms
+        long _lastReceivedLsn;
+        long _lastFlushedLsn;
+        long _lastAppliedLsn;
+
+        #endregion Fields
+
+        private protected NpgsqlReplicationConnection()
+        {
+            _npgsqlConnection = new NpgsqlConnection();
+            _requestFeedbackInterval = new TimeSpan(_walReceiverTimeout.Ticks / 2);
+            _sendFeedbackTimer = new Timer(TimerSendFeedback);
+            _requestFeedbackTimer = new Timer(TimerRequestFeedback);
+        }
+
+        #region Properties
+
+        /// <summary>
+        /// Gets or sets the string used to connect to a PostgreSQL database. See the manual for details.
+        /// </summary>
+        /// <value>The connection string that includes the server name,
+        /// the database name, and other parameters needed to establish
+        /// the initial connection. The default value is an empty string.
+        /// </value>
+        /// <remarks>
+        /// Since replication connections are a special kind of connection, the values for
+        /// Pooling, Enlist and Multiplexing are always false no matter what you set them to
+        /// in your connection string.
+        /// </remarks>
+        [AllowNull]
+        public string ConnectionString {
+            get => _userFacingConnectionString ?? string.Empty;
+            set
+            {
+                _userFacingConnectionString = value;
+                _npgsqlConnection.ConnectionString = new NpgsqlConnectionStringBuilder(value)
+                {
+                    Pooling = false,
+                    Enlist = false,
+                    Multiplexing = false,
```

Comment:
I know, but that might change at some point and replication connections will never be compatible with multiplexing so I prefer being on the safe side.

Reply:
Then maybe better to just throw if multiplexing is on. This way if the user explicitly puts it in the connection string they know it's not supported rather than believing they're using a multiplexing replication connection.

Reply:
> Then maybe better to just throw if multiplexing is on.

Wouldn't that be weird if, at some point, we'd decide to enable Multiplexing by default?
The user would have to explicitly disable it to avoid an exception.
Also Multiplexing users would need a different connection string for their replication connections (e. g. all of our current tests).
The way I see it is that enabling Multiplexing is a no-op in the replication context because that's definitely not what you want. We're talking about a single physical connection that lives for months if nothing goes wrong.

Reply:
> Wouldn't that be weird if, at some point, we'd decide to enable Multiplexing by default?

I really doubt that this will ever happen. 

But OK, we can just leave it like this.